src/binary_agreement/binary_agreement.rs

```diff
@@ -321,24 +317,19 @@ impl<N: NodeIdT> BinaryAgreement<N> {
 
     /// Creates the initial coin state for the current epoch, i.e. sets it to the predetermined
     /// value, or initializes a `ThresholdSign` instance.
-    fn coin_state(&self) -> CoinState<N> {
-        match self.epoch % 3 {
+    fn coin_state(&self) -> Result<CoinState<N>> {
+        Ok(match self.epoch % 3 {
             0 => CoinState::Decided(true),
             1 => CoinState::Decided(false),
             _ => {
-                let nonce = Nonce::new(
-                    self.netinfo.invocation_id().as_ref(),
-                    self.session_id,
-                    self.netinfo.node_index(&self.proposer_id).unwrap(),
-                    self.epoch,
-                );
-                CoinState::InProgress(Box::new(ThresholdSign::new(self.netinfo.clone(), nonce)))
+                let coin_id = bincode::serialize(&(&self.session_id, self.epoch))?;
```

Comment:
Did we sidestep the issue by just using an anonymous type in the form of a tuple to get the epoch info in there?